src/libcollections/borrow.rs

```diff
@@ -178,6 +178,30 @@ impl<'a, B: ?Sized> Cow<'a, B> where B: ToOwned {
             Owned(owned) => owned,
         }
     }
+
+    /// Makes a shallow copy of the data.
+    ///
+    /// Reborrows the data if it is already owned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(cow_shallow_copy)]
+    /// use std::borrow::Cow;
+    ///
+    /// let cow: Cow<[_]> = Cow::Owned(vec![1, 2, 3]);
+    /// match cow.shallow_copy() {
+    ///     Cow::Owned(_) => panic!("needless clone!"),
+    ///     Cow::Borrowed(vec) => assert_eq!(vec, &[1, 2, 3]),
+    /// }
+    /// ```
+    #[unstable(feature = "cow_shallow_copy", issue="0")]
+    pub fn shallow_copy(&'a self) -> Cow<'a, B> {
```

Comment:
Using `'a` like this is usually a red flag that deserves some scrutiny (when `'a` is the parameter used in the type, and not a method level parameter). I think `fn shallow_copy(&self) -> Cow<B>` would be a more natural